linode/acceptance/tmpl/linode_cloud_firewall.gotf

```diff
@@ -0,0 +1,38 @@
+{{ define "linode_cloud_firewall" }}
+
+data "http" "ipv6_addr" {
+  url = "https://api64.ipify.org?format=text"
+}
+
+data "http" "ipv4_addr" {
+  url = "https://api.ipify.org?format=text"
+}
+
+locals {
+  valid_ipv4_pattern = "^(25[0-5]|2[0-4][0-9]|[0-1]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[0-1]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[0-1]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[0-1]?[0-9][0-9]?)$$"
+  valid_ipv6_pattern = "^(?:[a-fA-F0-9]{1,4}:){7}[a-fA-F0-9]{1,4}|(?:[a-fA-F0-9]{1,4}:){1,7}:|(?:[a-fA-F0-9]{1,4}:){1,6}:[a-fA-F0-9]{1,4}|(?:[a-fA-F0-9]{1,4}:){1,5}:(?:[a-fA-F0-9]{1,4}:){1,4}|(?:[a-fA-F0-9]{1,4}:){1,4}:(?:[a-fA-F0-9]{1,4}:){1,4}|(?:[a-fA-F0-9]{1,4}:){1,3}:(?:[a-fA-F0-9]{1,4}:){1,4}|(?:[a-fA-F0-9]{1,4}:){1,2}:(?:[a-fA-F0-9]{1,4}:){1,4}|[a-fA-F0-9]{1,4}:(?:[a-fA-F0-9]{1,4}:){1,4}|(?:[a-fA-F0-9]{1,4}:){1,4}:|:(?::[a-fA-F0-9]{1,4}){1,7}|::|(?:[a-fA-F0-9]{1,4}:){1,7}(?::[a-fA-F0-9]{1,4}){1,7}$$"
+  valid_ipv4 = can(regex(local.valid_ipv4_pattern, data.http.ipv4_addr.response_body))
+  valid_ipv6 = can(regex(local.valid_ipv6_pattern, data.http.ipv6_addr.response_body))
+  ipv4_address = local.valid_ipv4 ? "${data.http.ipv4_addr.response_body}/32" : null
+  ipv6_address = local.valid_ipv6 ? "${data.http.ipv6_addr.response_body}/128" : null
+}
+
+resource "linode_firewall" "linode_cloud_firewall" {
+  label           = "{{.Label}}"
+  outbound_policy = "ACCEPT"
+  inbound_policy  = "DROP"
+
+  dynamic "inbound" {
```

Comment:
This incoming block will include either IPv4, IPv6, or both, depending on their validity and availability.